base/reduce.jl

```diff
@@ -360,15 +360,19 @@ sum(a::AbstractArray{Bool}) = countnz(a)
 Returns the sum of all array elements, using the Kahan-Babuska-Neumaier compensated
 summation algorithm for additional accuracy.
 """
-function sum_kbn{T<:AbstractFloat}(A::AbstractArray{T})
+function sum_kbn(A::Any)
```

Comment:
I sometimes use `::Any` if I want to emphasize, that a function literally accepts any input. But here this is not the case, an iterator is required, so I find the `::Any` slightly confusing.